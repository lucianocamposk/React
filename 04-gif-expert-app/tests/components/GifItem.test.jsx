import { render, screen } from '@testing-library/react';    
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GitItem />', () => { 

        const title = 'saitama';
        const url = 'https://one-punch.com/saitama.jpg'

        test('debe hacer match con el snapshot', () => { 
            const { container } = render( <GifItem title=' title ' url=' url ' />)
            expect ( container ).toMatchSnapshot();
         });
 
         
         test('debe de mostrar la imagen con la URL y el aLT indicado', () => {
            
            render( <GifItem title={ title } url={ url } />);
            // screen.debug();
            // expect( screen.getByRole('img').src ).toBe( url );
            const { src, alt } = screen.getByRole('img');
            expect ( src ).toBe( url );
            expect( alt ).toBe( alt );
        });

        test('debe de mostrar el titulo en el componenente', () => {

            render( <GifItem title={ title } url={ url } />);
            expect( screen.getByText( title ) ).toBeTruthy();

        });
});

