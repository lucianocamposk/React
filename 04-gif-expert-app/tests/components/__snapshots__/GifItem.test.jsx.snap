// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt=" title "
      src=" url "
    />
    <p>
       title 
    </p>
  </div>
</div>
`;
